src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/PolymorphicTests.CustomTypeHierarchies.cs

```diff
@@ -1762,12 +1880,34 @@ public async Task Peano_Deserialization(int expectedSize, string json)
             await TestMultiContextDeserialization<Peano>(json, expected);
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(4)] // initial Write/ReadStack size
+        [InlineData(5)]
+        [InlineData(63)]
+        [InlineData(64)] // initial Write/ReadStack size * 16
+        [InlineData(65)]
+        [InlineData(150)]
```

Comment:
That number is kind of large, suspect it's edging on causing SO in certain platforms.
```suggestion
```